### PR TITLE
Feature/timer

### PR DESCRIPTION
**Pull request by dtrunov**
Create timer based on python timer and add UT for timer.

### CODE REVIEW

File: pylua/test/test_timer.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+import pylua.timer
+import logging
+import time
+import threading
+
+logger = logging.getLogger(__name__)
+
+class TestPyTimer:
+	__test_timer = None
```
Comment (malirod) [inline pylua/test/test_timer.py]: @dtrunov , please use single leading underscore. you're no going to have subclasses, so no need in name mangling which is enabled by two leading underscores. One underscore is already a conventional sign of implementation details. On my taste two leading underscores loocs not good.

Please use 4 spaces instead of tabs.

Are you really sure that this is the right place for these vars? Vars declared this way are static ( http://stackoverflow.com/questions/68645/static-class-variables-in-python )
I think all these vars should not be static. some common stuff must be initiated in ctor (**init**), and the rest of timers have to be declared\initiated directly in test cases where they used.

File: pylua/test/test_timer.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	__start_timer = None
+	__finish_timer = None
+	__interval = None
+	__cond_var = None
+
+	def timer_task(self):
```
Comment (malirod) [inline pylua/test/test_timer.py]: @dtrunov, seems this is implementation detail method. Add one leading underscore.

File: pylua/test/test_timer.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+		self.__cond_var.acquire()
+		self.__finish_timer = float(time.time())
+		self.__cond_var.notify()
+		self.__cond_var.release()
+		pass
```
Comment (malirod) [inline pylua/test/test_timer.py]: @dtrunov, `pass` is a special keyword used to handle case when you have empty body function\method\loop\if . Since all your methods have body, please remove all `pass`

File: pylua/test/test_timer.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+	def run_cond_var(self, wait_time):
+		self.__cond_var.acquire()
+		self.__cond_var.wait(wait_time)
+		self.__cond_var.release()
+		pass
+
+	def test_run_timer(self, interval = None):
+		logger.debug('test_run_timer_for_one_sec')
```
Comment (malirod) [inline pylua/test/test_timer.py]: @dtrunov, please remove all log lines from the tests. Use logging only for debugging and then delete logs. By the way, such log line must be written by test framework.

File: pylua/timer.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+
+import threading
+
+class PyTimer:
+	__timer = None
```
Comment (malirod) [inline pylua/timer.py]: @dtrunov, move these static vars to ctor. use single leading underscore. Use 4 spaces instead of tabs.

File: pylua/timer.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+	def reset(self):
+		self.__timer.cancel()
+		self.create()
+		self.__timer.start()
+
+	def interval(self):
+		return self.__interval
+
+	def set_interval(self, interval):
```
Comment (malirod) [inline pylua/timer.py]: @dtrunov, please use properties : http://stackoverflow.com/questions/6618002/python-property-versus-getters-and-setters . to make this work change the class declaration `class PyTimer:` -> `class PyTimer(object):`

File: pylua/timer.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+# -*- coding: utf-8 -*-
+
+import threading
+
+class PyTimer:
+	__timer = None
+	__task = None
+	__interval = None
+
+	def __init__(self, task, interval):
```
Comment (malirod) [inline pylua/timer.py]: @dtrunov, seems we need "one shot" and repeatable timer. Here I see only one (not sure exactly what kind).

File: pylua/test/test_timer.py [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+		pass
+
+	def __init__(self):
+		self.__interval = 1
+		self.__test_timer = timer.PyTimer(self.timer_task, self.__interval)
+		self.__cond_var = threading.Condition()
+		pass
+
+	def run_cond_var(self, wait_time):
+		self.__cond_var.acquire()
```
Comment (malirod) [inline pylua/test/test_timer.py]: @dtrunov, in python usually you don't need all those cond vars etc. there is no real multi-threading because of GIL ( https://habrahabr.ru/post/149420/ ) . To wait for something use `threading.Event()`  (link has example) . I think you don't need this method at all. Just use events in-place.

File: pylua/test/test_timer.py [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+		self.__cond_var.wait(wait_time)
+		self.__cond_var.release()
+		pass
+
+	def test_run_timer(self, interval = None):
+		logger.debug('test_run_timer_for_one_sec')
+		self.__finish_timer = None
+		self.__start_timer = float(time.time())
```
Comment (malirod) [inline pylua/test/test_timer.py]: @dtrunov, see http://stackoverflow.com/questions/15707056/get-time-of-execution-of-a-block-of-code-in-python-2-7 
propose to use the top rated answer.

Please, rename `__start_timer` ->  `_started`, `__finish_timer` -> `_stopped`, `__test_timer` ->`_timer`. Currently names are confusing. You mixed time intervals and timer object

File: pylua/test/test_timer.py (hunk 4 above)
Comment (malirod) [inline pylua/test/test_timer.py]: @dtrunov, seconds is too big resolution. Please, use milliseconds, e.g. 20 or 50.

File: pylua/test/test_timer.py [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+		if (interval is None):
+			wait_time = self.__interval + 1
+			self.__test_timer.start()
+		else:
+			wait_time = interval + 1
+			self.__interval = interval
+			self.__test_timer.start(interval)
+		self.run_cond_var(wait_time)		    
+		assert isinstance(self.__finish_timer, float), "Timer doesn't work"
```
Comment (malirod) [inline pylua/test/test_timer.py]: @dtrunov, I think it's better to check `self._finished is not None`. This comparison with type confuses me

File: pylua/test/test_timer.py [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+		timer_infelicity = abs(self.__finish_timer - self.__start_timer - self.__interval)
```
Comment (malirod) [inline pylua/test/test_timer.py]: @dtrunov, what does `timer_infelicity` means? I don't understand the word.

File: pylua/test/test_timer.py [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+		self.__cond_var = threading.Condition()
+		pass
+
+	def run_cond_var(self, wait_time):
+		self.__cond_var.acquire()
+		self.__cond_var.wait(wait_time)
+		self.__cond_var.release()
+		pass
+
+	def test_run_timer(self, interval = None):
```
Comment (malirod) [inline pylua/test/test_timer.py]: @dtrunov, this is private method. add leading underscore to not mix this method with test cases for test framework.
I think it's even better to rename to `run_timer`

File: pylua/test/test_timer.py [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+		logger.debug('timer_infelicity = ')
+		assert timer_infelicity < 0.05, "Timer doesn't work correctly"
+
+	def test_start_timer_with_parameter(self):
+		logger.debug('test_start_timer_with_parameter')
+		self.test_run_timer(2)
+		pass
+	
+	def test_start_timer_periodicaly_with_parameter(self):
+		logger.debug('test_start_timer_periodicaly_with_parameter')
```
Comment (malirod) [inline pylua/test/test_timer.py]: @dtrunov, periodical timer must be a feature of the PyTimer itself.

File: pylua/test/test_timer.py [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+			self.__test_timer.start()
+		else:
+			wait_time = interval + 1
+			self.__interval = interval
+			self.__test_timer.start(interval)
+		self.run_cond_var(wait_time)		    
+		assert isinstance(self.__finish_timer, float), "Timer doesn't work"
+		timer_infelicity = abs(self.__finish_timer - self.__start_timer - self.__interval)
+		logger.debug('timer_infelicity = ')
+		assert timer_infelicity < 0.05, "Timer doesn't work correctly"
```
Comment (malirod) [inline pylua/test/test_timer.py]: @dtrunov, propose to change validation to

```
elapsed = self._stopped - self._started
assert elapsed >= interval
```

File: pylua/test/test_timer.py [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+	
+	def test_start_timer_periodicaly_with_parameter(self):
+		logger.debug('test_start_timer_periodicaly_with_parameter')
+		for i in range(0,5):
+			self.test_run_timer(2)
+		pass
+
+	def test_start_timer_periodicaly_without_parameter(self):
+		logger.debug('test_start_timer_periodicaly_without_parameter')
+		for i in range(0,5):
```
Comment (malirod) [inline pylua/test/test_timer.py]: @dtrunov , what is the difference in `test_start_timer_periodicaly_with_parameter` and `test_start_timer_periodicaly_without_parameter`? On my opinion they exactly the same but with different params. I don't like idea to run timers in a loop at all. I don't see the point - you're testing same thing many times.
Here are test cases I see:
1) lunch one shot timer
2) lunch repeatable timer, catch second iteration and break
3) start timer with some big interval -> don't wait and stop.

File: pylua/test/test_timer.py (hunk 12 above)
Comment (malirod) [reply pylua/test/test_timer.py]: @dtrunov, I don't see the point in this method at all. Create timers in test cases and write timer tasks as internal functions.

```
def test_case(self):
  def task():
     finished = now()
  started =now()
  timer = create_timer(task, some_small_interval)
  elapse = finished - started
  assert elapse >= some_small_interval
```

File: pylua/test/test_timer.py [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+			self.test_run_timer()
+		pass
+
+	def test_timer_set_interval(self):
+		logger.debug('test_timer_set_interval')
+		interval = 6
+		self.__test_timer.set_interval(interval)
```
Comment (malirod) [inline pylua/test/test_timer.py]: @dtrunov, I see no point in private vars. Use local vars

**Review by malirod — CHANGES_REQUESTED** [review]

**Comment by malirod** [comment]
@dtrunov, for some reason you PR work incorrectly on CI. Check log output ( https://travis-ci.org/malirod/pylua/builds/169981819 )
There are no style checks. There is only `No commits provided` line. Have you installed pre-commit hook locally, does it work for you?

Another very strange thing is that there are no timer tests in in the output. Your tests were not executed at all. Have you seen you tests in nosetests output locally?

File: pylua/test/test_timer.py (hunk 1 above)
Comment (dtrunov) [reply pylua/test/test_timer.py]: @malirod I will do this with next commit

File: pylua/test/test_timer.py (hunk 2 above)
Comment (dtrunov) [reply pylua/test/test_timer.py]: @malirod I will do this within next commit

File: pylua/test/test_timer.py (hunk 3 above)
Comment (dtrunov) [reply pylua/test/test_timer.py]: @malirod I will do this within next commit

File: pylua/test/test_timer.py (hunk 8 above)
Comment (dtrunov) [reply pylua/test/test_timer.py]: @malirod I have used event, but I renamed this method, see within next commit.

File: pylua/test/test_timer.py (hunk 12 above)
Comment (dtrunov) [reply pylua/test/test_timer.py]: @malirod I will rename this method and will rework it to private method. Please see this within next commit.

Comment (dtrunov) [reply pylua/test/test_timer.py]: @malirod Why can not I use external task in test case? Please explain me why this way better than my.

File: pylua/test/test_timer.py (hunk 4 above)
Comment (dtrunov) [reply pylua/test/test_timer.py]: @malirod I will do this within next commit. Please see.

File: pylua/test/test_timer.py (hunk 12 above)
Comment (malirod) [reply pylua/test/test_timer.py]: @dtrunov, if there will be single task for many test cases, then it's good to have some external private method for task. But in this case, after you'll rework tests, I think there will be no similarity. I think there will be diffident  tasks for each test case and in this case local function for task is better since in this case test case logic is localized. Plus you don't have to imagine unique name for each task, you can use `task` name for all.

**Comment by OHerasym** [comment]
@dtrunov , @malirod , added set_task method https://github.com/OHerasym/pylua/blob/feature/arg_parser/pylua/timer.py]

lua wrapper for timer: https://github.com/OHerasym/pylua/blob/feature/arg_parser/pylua/timer.lua

File: pylua/test/test_timer.py (hunk 12 above)
Comment (dtrunov) [reply pylua/test/test_timer.py]: @malirod I have other test cases: test_stop_timer, test_reset_timer, which use timer with external task, and If I rework task to internal, I have to use internal task for test cases: test_stop_timer, test_reset_timer. And I think  to rework task to internal method is bad way

Comment (malirod) [reply pylua/test/test_timer.py]: @dtrunov, please, fix the rest of comments, then there will be clear code and we'll see how to proceed.

File: pylua/test/test_timer.py (hunk 4 above)
Comment (dtrunov) [reply pylua/test/test_timer.py]: @malirod I will do this within next commit. Please see next commit

File: pylua/test/test_timer.py (hunk 9 above)
Comment (dtrunov) [reply pylua/test/test_timer.py]: @malirod I will do this within next commit. Please see next commit

File: pylua/test/test_timer.py (hunk 10 above)
Comment (dtrunov) [reply pylua/test/test_timer.py]: @malirod I removed this checking, please see this changes with next commit.

File: pylua/test/test_timer.py (hunk 11 above)
Comment (dtrunov) [reply pylua/test/test_timer.py]: @malirod I calculate some delay, which appears during run of timer. I test a lot of time and on my computer this delay equal = 50msec. I renamed this variable please see within next commit.

File: pylua/test/test_timer.py (hunk 13 above)
Comment (dtrunov) [reply pylua/test/test_timer.py]: @malirod I checked using of timer from lua, and I saw that timer is used as single shot. I add functionality to PyTimer that allows to run timer several times from one object.

File: pylua/test/test_timer.py (hunk 15 above)
Comment (dtrunov) [reply pylua/test/test_timer.py]: @malirod Please read comment above about single shot timer. I checked in this test cases functionality that I described  above. I checked test case 1 and 3.

File: pylua/test/test_timer.py (hunk 16 above)
Comment (dtrunov) [reply pylua/test/test_timer.py]: @malirod I try to save interface of QT timer that is used from lua.

File: pylua/timer.py (hunk 5 above)
Comment (dtrunov) [reply pylua/timer.py]: @malirod I will do this within next commit.

File: pylua/timer.py (hunk 7 above)
Comment (dtrunov) [reply pylua/timer.py]: @malirod I investigated lua code and I saw that timer is used as single shot.

File: pylua/timer.py (hunk 6 above)
Comment (dtrunov) [reply pylua/timer.py]: @malirod I will do this please see next commit.

**Comment by dtrunov** [comment]
@malirod Please review latest changes.

File: pylua/timer.py [hunk 17]
```diff
@@ -0,0 +1,48 @@
...
+
+class PyTimer(object):
+
+    def __init__(self, task, interval):
+        self._task = task
+        self._interval = convert_milliseconds_to_seconds(interval)
+        self._timer = None
+
+    def start(self, interval=None):
+        if interval is not None:
```
Comment (malirod) [inline pylua/timer.py]: @dtrunov, why do you need two methods (ctor, start) which allow to set interval? On my opinion this is redundant. Keep only ctor. If someone will need to change interval then timer can be recreated. This way test will be simpler. I see no point in logic duplication..

File: pylua/timer.py [hunk 18]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+# timer works with milliseconds
+
+import threading
+
+
+def convert_milliseconds_to_seconds(milliseconds):
```
Comment (malirod) [inline pylua/timer.py]: @dtrunov, on my taste to long name. This can be renamed to msec_to_sec and vise versa.

File: pylua/test/test_timer.py [hunk 19]
```diff
@@ -0,0 +1,84 @@
...
+    def _run_event(self, wait_time):
+        wait_time = timer.convert_milliseconds_to_seconds(wait_time)
+        self._event.clear()
+        self._event.wait(wait_time)
+
+    def _run_timer(self, interval=None):
+        self._task_is_run = False
+        started = timeit.default_timer()
+        if (interval is None):
+            wait_time = self._interval + 500
```
Comment (malirod) [inline pylua/test/test_timer.py]: @dtrunov. why 500? can we use 100?

File: pylua/test/test_timer.py [hunk 20]
```diff
@@ -0,0 +1,84 @@
...
+        self._task_is_run = True
+        self._event.set()
+
+    def __init__(self):
+        self._interval = 50  # interval in milliseconds
+        self._timer = timer.PyTimer(self._timer_task, self._interval)
+        self._task_is_run = False
+        self._event = threading.Event()
+
+    def _run_event(self, wait_time):
```
Comment (malirod) [inline pylua/test/test_timer.py]: @dtrunov, propose to rename `_run_event` -> `_wait_event` since you're not running anything here.

File: pylua/test/test_timer.py [hunk 21]
```diff
@@ -0,0 +1,84 @@
...
+        self._run_timer(50)  # interval in milliseconds
+
+    def test_start_timer_without_parameter(self):
+        self._run_timer()
+
+    def test_start_timer_periodicaly_with_parameter(self):
+        for i in range(0, 5):
+            self._run_timer(50)  # interval in milliseconds
+
+    def test_start_timer_periodicaly_without_parameter(self):
```
Comment (malirod) [inline pylua/test/test_timer.py]: @dtrunov, if you'll remove interval from `start` then this test case will be redundant. `test_start_timer_without_parameter` too

File: pylua/timer.py [hunk 22]
```diff
@@ -0,0 +1,48 @@
...
+    def stop(self):
+        self._timer.cancel()
+
+    def reset(self):
+        self._timer.cancel()
+        self.create()
+        self._timer.start()
+
+    @property
+    def interval(self):
```
Comment (malirod) [inline pylua/timer.py]: @dtrunov, why you need to expose interval propose? What are use cases? On my opinion this property and parameter in start are redundant. The possibility to change interval makes logic more complicated.

File: pylua/test/test_timer.py [hunk 23]
```diff
@@ -0,0 +1,84 @@
...
+        for i in range(0, 5):
+            self._run_timer()
+
+    def test_timer_set_interval(self):
```
Comment (malirod) [inline pylua/test/test_timer.py]: @dtrunov, am I right that you're testing python property feature? Why do think this is required?

File: pylua/test/test_timer.py [hunk 24]
```diff
@@ -0,0 +1,84 @@
...
+        self._timer.start(interval)
+        self._run_event(wait_time)
+        reseted = timeit.default_timer()
+        self._timer.reset()
+        wait_time = 1000
+        self._run_event(wait_time)
+        assert self._task_is_run is True, "Reset doesn't work"
+        delay = (timeit.default_timer() - reseted -
+                 timer.convert_milliseconds_to_seconds(interval))
+        assert delay < 0.06, "Timer doesn't work correctly"
```
Comment (malirod) [inline pylua/test/test_timer.py]: @dtrunov, why `0.06`? Why not `0.07` or `0.05`? I've proposed you the following assert: `assert delay finish_time -start_time < msec_to_sec(interval)`. You don't like this?  Set start_time before _timer.start, finish_time set after _wait_even

**Review by malirod — CHANGES_REQUESTED** [review]

**Comment by malirod** [comment]
@dtrunov, please rebase the PR after all comments will be addressed somehow(otherwise they will be lost). I've fixed style checker and enabled it on CI.

I've tested you're branch locally and there were some warnings from pylint (timer test).

Check readme in develop to see ho to use style checker locally. actually it should be used automatically by pre-commit hook, but seems during rebase hook will not fire, so I've described the way how to test commits range to test whole branch at once.

File: pylua/timer.py (hunk 18 above)
Comment (dtrunov) [reply pylua/timer.py]: @malirod I will do it within next commit.

File: pylua/timer.py (hunk 17 above)
Comment (dtrunov) [reply pylua/timer.py]: @malirod I will do it with next commit

File: pylua/timer.py (hunk 22 above)
Comment (dtrunov) [reply pylua/timer.py]: @malirod I will do this within next commit

File: pylua/test/test_timer.py (hunk 20 above)
Comment (dtrunov) [reply pylua/test/test_timer.py]: @malirod I will do it within next commit

File: pylua/timer.py (hunk 22 above)
Comment (dtrunov) [reply pylua/timer.py]: @malirod I will do this with next commit

File: pylua/test/test_timer.py (hunk 20 above)
Comment (dtrunov) [reply pylua/test/test_timer.py]: @malirod I will do this with next commit

File: pylua/test/test_timer.py (hunk 19 above)
Comment (dtrunov) [reply pylua/test/test_timer.py]: @malirod I will do this with next commit

File: pylua/test/test_timer.py (hunk 21 above)
Comment (dtrunov) [reply pylua/test/test_timer.py]: @malirod I will do this with next commit

File: pylua/test/test_timer.py (hunk 23 above)
Comment (dtrunov) [reply pylua/test/test_timer.py]: @malirod I will do this with next commit

File: pylua/test/test_timer.py (hunk 24 above)
Comment (dtrunov) [reply pylua/test/test_timer.py]: @malirod I will do this with next commit

**Comment by dtrunov** [comment]
@malirod Please review latest changes

**Comment by malirod** [comment]
@dtrunov, pylint has some issue on Travis. But locally I've got the following output on your branch

```
 Checking python style with pylint 
************* Module pylua.test.test_timer
C: 34, 0: Line too long (117/100) (line-too-long)
C: 54, 0: Exactly one space required after assignment
        self._timer =  timer.PyTimer(self._timer_task, interval)
                    ^ (bad-whitespace)
C: 63, 0: Line too long (111/100) (line-too-long)
C: 10, 0: Old-style class defined. (old-style-class)
W: 40,12: Unused variable 'i' (unused-variable)
 Pylint reports about the issues in the python scripts 

```

File: pylua/timer.py [hunk 25]
```diff
@@ -0,0 +1,38 @@
...
+        self._task = task
+        self._interval = msec_to_sec(interval)
+        self._timer = None
+
+    def start(self):
+        self.create()
+        self._timer.start()
+
+    def create(self):
+        self._timer = threading.Timer(self._interval, self._task)
```
Comment (malirod) [inline pylua/timer.py]: @dtrunov, why you need a special creator for the timer? Why not just to use ctor? You're just duplicating logic of the ctor in `create`. With this special factory method you have to handle `_timer is None` in rest of methods. I think it's better to drop `create`.

File: pylua/test/test_timer.py [hunk 26]
```diff
@@ -0,0 +1,63 @@
...
+
+    def __init__(self):
+        self._interval = 50  # interval in milliseconds
+        self._timer = timer.PyTimer(self._timer_task, self._interval)
+        self._task_is_run = False
+        self._event = threading.Event()
+
+    def _wait_event(self, wait_time):
+        wait_time = timer.msec_to_sec(wait_time)
+        self._event.clear()
```
Comment (malirod) [inline pylua/test/test_timer.py]: @dtrunov, call `clear` in `_run_timer` before `self._timer.start()`. In general it's possible that your timer will finish before wait event will enter. So wait will never end.

File: pylua/timer.py [hunk 27]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+# timer works with milliseconds
```
Comment (malirod) [inline pylua/timer.py]: @dtrunov, this is wrong place for documentation. Use doxygen.
See python sample here http://www.stack.nl/~dimitri/doxygen/manual/docblocks.html

File: pylua/test/test_timer.py [hunk 28]
```diff
@@ -0,0 +1,63 @@
...
+        self._task_is_run = True
+        self._event.set()
+
+    def __init__(self):
+        self._interval = 50  # interval in milliseconds
+        self._timer = timer.PyTimer(self._timer_task, self._interval)
+        self._task_is_run = False
+        self._event = threading.Event()
+
+    def _wait_event(self, wait_time):
```
Comment (malirod) [inline pylua/test/test_timer.py]: @dtrunov, propose to rename `wait_time` -> `timeout`

**Review by malirod — CHANGES_REQUESTED** [review]

File: pylua/timer.py (hunk 25 above)
Comment (dtrunov) [reply pylua/timer.py]: @malirod I use create because I can not start timer which finised work again  For this reason need to recreate timer.

File: pylua/test/test_timer.py (hunk 26 above)
Comment (dtrunov) [reply pylua/test/test_timer.py]: @malirod Why? Event has its own timeout and as far as I understand if nobody call event.set() main thread continues to work after event timeout.

File: pylua/timer.py (hunk 25 above)
Comment (malirod) [reply pylua/timer.py]: @dtrunov, why this create methon cannot be replaced with ctor? I see no point in special method to create timer.

File: pylua/test/test_timer.py (hunk 26 above)
Comment (malirod) [reply pylua/test/test_timer.py]: @ dtrunov, yes, python impl of event has a timeout and if for some reason there will problem I've described, then after timeout wait will be unblocked. But it's  icorrect to rely o. Timeout in this case. There is common pattern, which apllicable to any impl. Stick with it. Usualy timeout is qute big and this some fallback logic to prevent a deadlock. SInce your  test doesnt chech edge case, please set timeoout here to some infinite value. I want with method to forever if something will go wrong. TRavis has1 h timeout as far as i know so after some time build will fail in the worst case.

File: pylua/timer.py (hunk 27 above)
Comment (dtrunov) [reply pylua/timer.py]: @malirod Fixed within b3bcde3

File: pylua/timer.py (hunk 25 above)
Comment (dtrunov) [reply pylua/timer.py]: @malirod Fixed within 6b7378a

File: pylua/test/test_timer.py (hunk 26 above)
Comment (dtrunov) [reply pylua/test/test_timer.py]: @malirod Fixed within 6b7378a

File: pylua/test/test_timer.py (hunk 28 above)
Comment (dtrunov) [reply pylua/test/test_timer.py]: @malirod fixed within b3bcde3

**Comment by dtrunov** [comment]
@malirod  Please review latest changes. I will squash all commits after finishing of review.

File: pylua/timer.py [hunk 29]
```diff
@@ -0,0 +1,38 @@
...
+Module implements wrapper of python
+timer. Wrapper allows to use timer with milliseconds
+"""
+
+import threading
+
+
+def msec_to_sec(milliseconds):
+    milliseconds_in_second = 1000.0
+    return milliseconds/milliseconds_in_second
```
Comment (malirod) [inline pylua/timer.py:15]: @dtrunov, please handle none param value in both helpers. I think assert is enough

**Review by malirod — CHANGES_REQUESTED** [review]

**Comment by malirod** [comment]
@dtrunov, I've left one minor comment. The rest looks good. You can squash

File: pylua/timer.py (hunk 29 above)
Comment (dtrunov) [reply pylua/timer.py:15]: @malirod  Fixed with commit 138e3f1

**Comment by dtrunov** [comment]
@malirod  Please review latest changes

**Review by malirod — APPROVED** [review]
